### ISSUE TITLE
Inconsistent documentation of default StorageClass

### ISSUE THREAD

**Issue by Nuru** [kubernetes/website#42288]
The documentation page "[Change the default Storage Class](https://kubernetes.io/docs/tasks/administer-cluster/change-default-storage-class/)" has a significant inconsistency regarding having more than one StorageClass marked as default.

"[Changing the default StorageClass -> Mark a StorageClass as default](https://kubernetes.io/docs/tasks/administer-cluster/change-default-storage-class/#changing-the-default-storageclass)" says:

> Please note that at most one StorageClass can be marked as default. If two or more of them are marked as default, a PersistentVolumeClaim without storageClassName explicitly specified cannot be created.

The above is wrong as far as I know, and it is also contradicted by [Storage Classes](https://kubernetes.io/docs/concepts/storage/storage-classes/#default-storageclass), which says:

> The cluster can only have one default StorageClass. If more than one default StorageClass is accidentally set, the newest default is used when the PVC is dynamically provisioned.

See also https://github.com/kubernetes/website/issues/42413

## Update

Also in need of updating:

- [Admission Controllers](https://v1-26.docs.kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#defaultstorageclass) says:

> This admission controller does not do anything when no default storage class is configured. When more than one storage class is marked as default, it rejects any creation of PersistentVolumeClaim with an error and an administrator must revisit their StorageClass objects and mark only one as default.

I believe this was correct for Kubernetes 1.25 (and several earlier versions) but is wrong as of Kubernetes 1.26. See https://github.com/kubernetes/kubernetes/pull/110559

- [Persistent Volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) says (with respect to the `DefaultStorageClass` admission plugin):

> If more than one default is specified, the admission plugin forbids the creation of all PVCs.

Which again I believe is wrong as of Kubernetes 1.26. 

## Summary of Affected pages

The following are links to the current documentation page and the v1.26 version of pages affected by the issue and therefore in need of updating. Likely all versions in between also need to be updated.

### Administration Tasks

- https://kubernetes.io/docs/tasks/administer-cluster/change-default-storage-class/
- https://v1-26.docs.kubernetes.io/docs/tasks/administer-cluster/change-default-storage-class/

### Admission Controllers

- https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#defaultstorageclass
- https://v1-26.docs.kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#defaultstorageclass

### Persistent volumes

- https://kubernetes.io/docs/concepts/storage/persistent-volumes/#class-1
- https://v1-26.docs.kubernetes.io/docs/concepts/storage/persistent-volumes/#class-1

**Comment by dipesh-rawat** [comment]
Pages reported in issue: https://kubernetes.io/docs/tasks/administer-cluster/change-default-storage-class/ , https://kubernetes.io/docs/concepts/storage/storage-classes/
/language en

**Comment by Nuru** [comment]
After further research, based on https://github.com/kubernetes/kubernetes/pull/110559, I believe the correct documentation is:

> Prior to Kubernetes 1.26, if more than one StorageClass is marked as default,
a PersistentVolumeClaim without `storageClassName` explicitly specified cannot be created.
In Kubernetes 1.26 and later, if more than one StorageClass is marked as default,
the last one created will be used.

Please have someone with detailed knowledge confirm, and then update the documentation accordingly.

**Comment by Affan-7** [comment]
It's a duplicate of #42176

**Comment by Nuru** [comment]
@dipesh-rawat Please amend this Issue to note the following pages are also affected:

### Admission Controllers

- https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#defaultstorageclass
- https://v1-26.docs.kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#defaultstorageclass

### Persistent volumes

- https://kubernetes.io/docs/concepts/storage/persistent-volumes/#class-1
- https://v1-26.docs.kubernetes.io/docs/concepts/storage/persistent-volumes/#class-1

@Affan-7 This is not a duplicate of #42176, it is a superset of it, and will not be completely fixed by #42177.

**Comment by Nuru** [comment]
@Affan-7 Please note that you are incorrect in marking this a duplicate of #42176. Although the subject matter is the same, the specific piece of documentation that needs to be revised is different, and still incorrect even after #42177 has been applied.

**Comment by Affan-7** [comment]
> @Affan-7 Please note that you are incorrect in marking this a duplicate of #42176. Although the subject matter is the same, the specific piece of documentation that needs to be revised is different, and still incorrect even after #42177 has been applied.

Sure I have deleted the comment.

**Comment by sftim** [comment]
Contributors are welcome to work on this issue. See https://k8s.io/docs/contribute/docs/ for a guide on getting started, if that's helpful.
